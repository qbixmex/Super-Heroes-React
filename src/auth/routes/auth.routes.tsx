import { Routes, Route, Navigate } from 'react-router-dom';
import { LoginPage } from '../pages';

export function AuthRoutes() {
  return (
    <Routes>
      <Route path='login' element={ <LoginPage /> } />
      <Route path='/*' element={ <Navigate to='/auth/login' /> } />
    </Routes>
  );
}