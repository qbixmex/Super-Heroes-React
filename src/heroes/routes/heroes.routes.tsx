import { Routes, Route } from 'react-router-dom';
import { Navigation } from '../../layout/navigation';
import { HeroesPage } from '../pages';
import { NotFoundPage } from '../pages/not.found.page';

export function HeroesRoutes() {
  return (
    <>
      <Navigation />
      <Routes>
        <Route path="/" element={<HeroesPage />} />
        <Route path="*" element={<NotFoundPage />} />
      </Routes>
    </>
  );
}
