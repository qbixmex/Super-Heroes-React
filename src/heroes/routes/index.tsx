export { HeroesRoutes } from './heroes.routes';
