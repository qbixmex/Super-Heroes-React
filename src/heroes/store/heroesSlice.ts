import { createSlice } from '@reduxjs/toolkit';
import type { PayloadAction } from '@reduxjs/toolkit';
import { Hero } from '../../interfaces';

export type HeroesState = {
  heroes: Hero[];
  isLoading?: boolean;
  activeHero?: null | Hero;
};

const initialState: HeroesState = {
  heroes: [],
  isLoading: false,
  activeHero: null,
};

export const heroesSlice = createSlice({
  name: 'heroes',
  initialState,
  reducers: {
    startLoadingHeroes: (state) => {
      state.isLoading = true;
    },
    setHeroes: (state, action: PayloadAction<HeroesState>) => {
      state.isLoading = false;
      state.heroes = action.payload.heroes;
    },
  },
});

export const { startLoadingHeroes, setHeroes } = heroesSlice.actions;
