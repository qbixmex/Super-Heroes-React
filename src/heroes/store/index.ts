export * from './heroesSlice';
export * from './store';
export * from './thunks';
