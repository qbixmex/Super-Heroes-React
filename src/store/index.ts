export * from './slices/heroesSlice';
export * from './store';
export * from './thunks';
