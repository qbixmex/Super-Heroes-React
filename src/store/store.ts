import { configureStore } from '@reduxjs/toolkit';
import { heroesSlice, usersSlice, authSlice } from './slices';

export const store = configureStore({
  reducer: {
    heroes: heroesSlice.reducer,
    users: usersSlice.reducer,
    auth: authSlice.reducer,
  },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;

// Inferred type: {heroes: HeroesState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
