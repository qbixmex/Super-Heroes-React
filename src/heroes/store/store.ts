import { configureStore } from '@reduxjs/toolkit';
import { heroesSlice } from './heroesSlice';

export const store = configureStore({
  reducer: {
    heroes: heroesSlice.reducer,
  },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;

// Inferred type: {heroes: HeroesState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
