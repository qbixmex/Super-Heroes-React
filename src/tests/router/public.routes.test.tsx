import { describe, test } from 'vitest';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { PublicRoutes } from '../../router/public.routes';

import { store } from '../../store/store';

describe('Test on <PublicRoutes />', () => {
  test('Should show default page', () => {
    const { container } = render(
      <Provider store={ store }>
        <MemoryRouter initialEntries={['/auth/login']}>
          <PublicRoutes />
        </MemoryRouter>
      </Provider>,
    );

    const loginForm = container.querySelector('#login-form');

    expect(loginForm).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test('Should redirect to login page', () => {
    const { container } = render(
      <Provider store={ store }>
        <MemoryRouter initialEntries={['/auth/wrong/page']}>
          <PublicRoutes />
        </MemoryRouter>
      </Provider>,
    );

    const loginForm = container.querySelector('#login-form');

    expect(loginForm).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
