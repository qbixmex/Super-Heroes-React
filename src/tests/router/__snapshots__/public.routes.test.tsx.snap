// Vitest Snapshot v1

exports[`Test on <PublicRoutes /> > Should redirect to login page 1`] = `
<div>
  <div
    class="login-page"
  >
    <div
      class="card bg-light text-dark border-light"
    >
      <div
        class="card-body"
      >
        <form
          id="login-form"
        >
          <div
            class="text-dark mb-3"
          >
            <label
              class="form-label"
              for="email"
            >
              Email
            </label>
            <input
              autocomplete="off"
              class="form-control"
              data-testid="email"
              id="email"
              name="email"
              type="email"
              value="stan-lee@marvel.com"
            />
          </div>
          <div
            class="text-dark mb-3"
          >
            <label
              class="form-label"
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="off"
              class="form-control"
              data-testid="password"
              id="password"
              name="password"
              type="password"
              value="secret-password"
            />
          </div>
          <div
            class="d-flex justify-content-end"
          >
            <button
              class="btn btn-secondary"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test on <PublicRoutes /> > Should show default page 1`] = `
<div>
  <div
    class="login-page"
  >
    <div
      class="card bg-light text-dark border-light"
    >
      <div
        class="card-body"
      >
        <form
          id="login-form"
        >
          <div
            class="text-dark mb-3"
          >
            <label
              class="form-label"
              for="email"
            >
              Email
            </label>
            <input
              autocomplete="off"
              class="form-control"
              data-testid="email"
              id="email"
              name="email"
              type="email"
              value="stan-lee@marvel.com"
            />
          </div>
          <div
            class="text-dark mb-3"
          >
            <label
              class="form-label"
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="off"
              class="form-control"
              data-testid="password"
              id="password"
              name="password"
              type="password"
              value="secret-password"
            />
          </div>
          <div
            class="d-flex justify-content-end"
          >
            <button
              class="btn btn-secondary"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
